sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -23,16 +23,21 @@ static Command *getCommand(const EventImplPtr &Event) {
 
 std::vector<EventImplPtr>
 Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
-  Command *Cmd = getCommand(Event);
-  // Command can be nullptr if user creates cl::sycl::event explicitly,
-  // as such event is not mapped to any SYCL task.
-  if (!Cmd)
-    return {};
   std::vector<EventImplPtr> Result;
-  for (const DepDesc &Dep : Cmd->MDeps) {
+
+  std::vector<DepDesc> Deps = Event->getDeps();
+  std::vector<EventImplPtr> PDeps = Event->getPreparedDepsEvents();
+  std::vector<EventImplPtr> PHDeps = Event->getPreparedHostDepsEvents();
```

Comment:
```suggestion
  const std::vector<DepDesc> &Deps = Event->getDeps();
  const std::vector<EventImplPtr> &PDeps = Event->getPreparedDepsEvents();
  const std::vector<EventImplPtr> &PHDeps = Event->getPreparedHostDepsEvents();
```
Otherwise an unnecessary copy is made.

Reply:
Done